Course video settings UI

mushtaqak opened the pull request:
This PR adds course video transcript UI pane on studio video uploads page.

[Sandbox link](https://studio-coursevideosettings.sandbox.edx.org/videos/course-v1:edX+DemoX+Demo_Course)

[EDUCATOR-1063](https://openedx.atlassian.net/browse/EDUCATOR-1063)

FYI @muzaffaryousaf

mushtaqak commented:
@muzaffaryousaf  @sstack22 This is how I am planning to build video uploads page. See attached snapshot.

<img width="1206" alt="screen shot 2017-08-07 at 11 55 52 pm" src="https://user-images.githubusercontent.com/6991154/29041371-0563cb22-7bcc-11e7-8a11-f39ccce27ed0.png">

and on clicking `Manage transcript languages` link, a popup/modal-like  UI would be displayed letting user to select a language from available transcription plan languages.

muzaffaryousaf commented:
sounds okay to me but let's run through roderick once he came in tomorrow. @sstack22 Let us know what do you think.

mushtaqak commented on requirements/edx/base.txt:
```diff
@@ -56,7 +56,6 @@ edx-organizations==0.4.5
 edx-rest-api-client==1.7.1
 edx-search==1.1.0
 edx-submissions==2.0.9
-edxval==0.1.1
```
Note: This is just for sandbox and for jenkins tests. Add this back this before merging PR.

mushtaqak commented on requirements/edx/github.txt:
```diff
@@ -92,6 +92,7 @@ git+https://github.com/edx/xblock-utils.git@v1.0.5#egg=xblock-utils==1.0.5
 git+https://github.com/edx/edx-user-state-client.git@1.0.1#egg=edx-user-state-client==1.0.1
 git+https://github.com/edx/xblock-lti-consumer.git@v1.1.5#egg=lti_consumer-xblock==1.1.5
 git+https://github.com/edx/edx-proctoring.git@0.18.2#egg=edx-proctoring==0.18.2
+git+https://github.com/edx/edx-val.git@1674ff42adcce7168f63acc16a44d2ef7f0e986f#egg=edxval==0.0.17
```
Note: This is just for sandbox and for jenkins tests. Remove this before merging PR.

mushtaqak commented on cms/static/js/views/active_video_upload_list.js:
```diff
@@ -59,6 +69,162 @@ define([
+                    this.selectedProvider = this.activeTranscriptionPlan['provider'];
+                    this.selectedFidelityPlan = this.activeTranscriptionPlan['cielo24_fidelity'];
+                    this.selectedTurnaroundPlan = this.activeTranscriptionPlan['cielo24_turnaround'] ? this.activeTranscriptionPlan['cielo24_turnaround']: this.activeTranscriptionPlan['three_play_turnaround'];
+                    this.activeLanguages = this.activeTranscriptionPlan['preferred_languages'];
```
**TODO**: May need to deserialise this as it is probably received like this format `"[u'ru', u'en']"`

mushtaqak commented:
@cptvitamin Can you  do a quick a11y review on course video settings (based on [basic mockups](http://jrpb2q.axshare.com/#g=1&p=snapshot-mvp)) ? You may need to click **course video settings** button on right side on Video Uploads page to view the course video settings pane.

FYI: @muzaffaryousaf @roderickmorales

mushtaqak commented:
@muzaffaryousaf Sandbox updated.

muzaffaryousaf commented:
jenkins run python

mushtaqak commented:
@roderickmorales @srpearce @sstack22 This is ready for your reviews :)

Sandbox: https://studio-coursevideosettings.sandbox.edx.org/videos/course-v1:edX+DemoX+Demo_Course

FYI: @muzaffaryousaf

srpearce commented:
OK, here are my suggestions for UI text.

**Button on Video Uploads page**

Current text | Suggested text
------------ | -------------
Course Video Settings | Video Transcript Settings

**Modal with "None" selected**

Current text | Suggested text
------------ | -------------
Course Video Settings | Video Transcript Settings
Transcript provider | Transcript Provider
3PlayMedia | 3Play Media
Transcript languages | Transcript Languages
Choose a language | Select language

**Modal with "3Play Media" selected**

Current text | Suggested text
------------ | -------------
Transcript turnaround | Transcript Turnaround
4-Day/Default | 4 days (default)
Same Day | Same day 
24-hour/Rush | 24 hours (rush)
10-Day/Extended | 10 days (extended)
2-Day/Expedited | 2 days (expedited)

Also, can we order the options other than "4 days (default)" according to duration? So it would be:

Same day
24 hours (rush)
2 days (expedited)
10 days (extended)

**Modal with "Cielo24" selected**

Current text | Suggested text
------------ | -------------
Transcript turnaround | Transcript Turnaround
Priority, 24h | Priority (24 hours)
Standard, 48h | Standard (48 hours)
Transcript fidelity | Transcript Fidelity
Professional, 99% Accuracy | Professional (99% accuracy)
Premium, 95% Accuracy | Premium (95% accuracy)
Mechanical, 75% Accuracy | Mechanical (75% accuracy)

mushtaqak commented:
>Current text | Suggested text | Reason
------------ | ------------- | -------------
Course Video Settings | Video Transcript Settings | @roderickmorales and I chose `Course Video Settings` because in future this settings pane may get  settings other than transcripts.

@srpearce Addressed you feedback. Rest of the texts are from edx-val repo. I am creating that PR and tag you there as well :)

mushtaqak commented on cms/static/cms/js/main.js:
```diff
@@ -64,7 +64,8 @@
                     data: JSON.stringify(data),
-                    success: callback
+                    success: callback,
+                    global: data ? data.global: true    // Trigger global AJAX error handler or not
```
*Note to reviewers*:  I have added this to support `global:false` in postJSON method.

mushtaqak commented on cms/static/js/views/course_video_settings.js:
```diff
@@ -0,0 +1,544 @@
+                        HtmlUtils.interpolateHtml(
+                            HtmlUtils.HTML('<div class="course-video-settings-success"><span class="icon fa fa-check-circle" aria-hidden="true"></span><span>{text}</span></div>'),
+                            {
+                                text: gettext('Settings updated.')
```
@srpearce can you also look at this string?

mushtaqak commented on cms/static/js/views/course_video_settings.js:
```diff
@@ -0,0 +1,544 @@
+                        HtmlUtils.interpolateHtml(
+                            HtmlUtils.HTML('<div class="course-video-settings-error"><span class="icon fa fa-info-circle" aria-hidden="true"></span><span>{text}</span></div>'),
+                            {
+                                text: errorMessage || gettext('Error saving data.')
```
@srpearce can you also look at this string?

mushtaqak commented:
@muhammad-ammar @muzaffaryousaf Can you do engineering review as well ?

roderickmorales commented:
@mushtaqak here is a spec to address spacing and some basic typography, and colors.
http://jrpb2q.axshare.com/specs.html

muzaffaryousaf commented:
@muhammad-ammar You can give it a shot first.

mushtaqak commented:
@roderickmorales @srpearce Address your feedbacks. Please check sandbox: https://studio-coursevideosettings.sandbox.edx.org/videos/course-v1:edX+DemoX+Demo_Course

FYI: @muzaffaryousaf

roderickmorales commented:
@mushtaqak it's looking good. Here is a little more feedback:

- "Transcript Languages" is not semibold
- Space between transcript languages and the first language is too big (should be about 16px)
- Background color, top and bottom border for languages odd rows need to be updated to reflect the specs (my bad, I missed those specs in the early specs document)
- When the user closes the "settings" div, it should slide out of the view

sstack22 commented:
Looks great to me, 👍 . Just a note on roll out: we should be able to enable this for individual courses, as we will test with Harvard first.

cptvitamin commented:
@mushtaqak I noticed some issues with the Course Video Settings "flyout."  This UI concept is sending mixed messages (both visually and programmatically).  It has components of both a disclosure widget (show/hide) and a modal dialog box.  We should pick one of those and then clean up the appearance and behaviors.  To me, this is more like a modal dialog box (since it has a close button). For disclosure widgets, the triggering button itself would toggle the display of the widget.  If this is a modal dialog box, we have some UX and accessibility work to do:

1. When the Course Video Settings button is clicked, the dialog should appear and focus needs to be moved to the first focusable element, which is the Close button.
2. Focus should remain inside the modal until ESC is pressed, or the Close button is clicked.
3. The modal must contain the following attributes: ```role=dialog aria-modal=true aria-label="Course Video Settings"```   (Since that text already exists in the button that triggered it, and that text is properly translated, you may choose to use aria-labelledby in lieu of aria-label.  aria-labelledby takes an IDref, so one would need to be available on the button.)


More info on dialog boxes here:
https://www.w3.org/TR/wai-aria-practices/#dialog_modal

cptvitamin commented on cms/templates/videos_index.html:
```diff
@@ -56,7 +59,7 @@ <h1 class="page-header">
             <h3 class="sr">${_("Page Actions")}</h3>
             <ul>
```
Can we get rid of the ```<ul> and <li>``` markup here?   A list of one is not really a list at all.  It adds so much extra verbosity and keystrokes for screen reader users.

cptvitamin commented:
@mushtaqak When using the keyboard to change the value of the Transcript Provider radio buttons, focus does not remain on the radio buttons.  Are you replacing the DOM node that contains the radio buttons?  That would cause this behavior.  If so, we need to  make sure we do not replace that DOM node, only the nodes below that are actually changing.  Also, the Update Settings button has important text that follows it.  Can we put an ID on that text and add an ```aria-describedby=[IDref]``` on that button?  That way the user is informed when it was last updated before they submit the form.

srpearce commented:
Just a couple of things.

**Button on Video Uploads page**

Current text | Suggested text
------------ | -------------
Course Video Settings | Transcript Settings

Reason: These only settings in this modal are transcript settings. "Course Video Settings" implies that this modal will have other video settings as well.

**Modal with "None" selected**

Current text | Suggested text
------------ | -------------
Course Video Settings | Transcript Settings

Reason: These only settings in this modal are transcript settings. "Course Video Settings" implies that this modal will have other video settings as well.

**Modal with "3Play Media" selected, Transcript Turnaround list**

Current text | Suggested text
------------ | -------------
10 Days (extended) | 10 days (extended) - note the lowercase d in "days".

Also, repeating this suggestion: **Can we order the options in this list according to duration?** We would leave "4 days (default)" as the top option. The options would be in this order:

4 days (default)
Same day
24 hours (rush)
2 days (expedited)
10 days (extended)

**Modal with "Cielo24" selected, Transcript Fidelity list**

Make all instances of "Accuracy" lowercase (accuracy).

cptvitamin commented:
@mushtaqak The contrast on the links used inside the dialog (REmove, ADd, etc) do not meet min contrast requirements.  Also, the select menus used to specify a language do not have labels.  Since the first value is an appropriate lable, you can use a little trick like this:
``` 
<select class="transcript-language-menu" id="transcript-language-menu-0" aria-labelledby="first">
  <option value="" id="first">Select language</option>
```

cptvitamin commented:
@mushtaqak the markup you used for the radio button group label is not quite correct.  The label "transcript provider" says its a label for "transcript-provider" but that is the ```name``` attribute on the radio buttons. It actually must refer to the container that wraps the radio buttons, so if you put an id that matches on the container div like this:
```<div class="transcript-provider-group" id=transcript-provider>``` 
you'll be all set.

muhammad-ammar commented on cms/static/js/views/course_video_settings.js:
```diff
@@ -0,0 +1,545 @@
+            // Clear error state if present any.
+            this.clearPreferanceErrorState($turnaroundContainer);
+
+            if(self.selectedProvider && turnaroundPlan) {
```
`if(turnaroundPlan)` is enough as `turnaroundPlan` is already checking `self.selectedProvider`

muhammad-ammar commented on cms/static/js/views/course_video_settings.js:
```diff
@@ -0,0 +1,545 @@
+            this.clearPreferanceErrorState($fidelityContainer);
+
+            // Fidelity dropdown
+            if (self.selectedProvider &&fidelityPlan) {
```
`if(fidelityPlan)` is enough as `fidelityPlan ` is already checking `self.selectedProvider`

muhammad-ammar commented on cms/static/sass/views/_video-upload.scss:
```diff
@@ -13,6 +13,143 @@
+      background-color: $gray-l3;
+      .action-close-course-video-settings {
+        width: 100%;
+        padding: 10px 20px;
```
please use $baseline

muhammad-ammar commented on cms/static/sass/views/_video-upload.scss:
```diff
@@ -13,6 +13,143 @@
+      margin-top: 30px;
+      .last-updated-text {
+        font-size: 0.75em;
+        margin-left: 5px;
```
mixin for rtl/ltr rules like `@include margin-left($baseline/4);`

muhammad-ammar commented on cms/static/sass/views/_video-upload.scss:
```diff
@@ -13,6 +13,143 @@
+          }
+          .remove-language-action {
+            display: inline-block;
+            float: right;
```
mixin

muhammad-ammar commented on cms/static/sass/views/_video-upload.scss:
```diff
@@ -13,6 +13,143 @@
+      .transcript-preferance-wrapper {
+        margin-top: 20px;
+        .icon.fa-info-circle {
+          margin-left: 5px;
```
mixin

muhammad-ammar commented on cms/static/sass/views/_video-upload.scss:
```diff
@@ -13,6 +13,143 @@
+        padding: 10px;
+        margin-bottom: 20px;
+        .icon {
+          margin-right: 5px;
```
mixin

muhammad-ammar commented on cms/static/sass/views/_video-upload.scss:
```diff
@@ -13,6 +13,143 @@
+    }
+
+    .course-video-settings-wrapper {
+      padding: 20px 20px;
```
same as above

muhammad-ammar commented on cms/static/sass/views/_video-upload.scss:
```diff
@@ -13,6 +13,143 @@
+
+      .course-video-settings-success, .course-video-settings-error {
+        padding: 10px;
+        margin-bottom: 20px;
```
$baseline for both above

muhammad-ammar commented on cms/static/sass/views/_video-upload.scss:
```diff
@@ -50,12 +187,6 @@
-
         .video-max-file-size-text {
           margin-top: ($baseline/2);
         }
```
Please use $baseline calculation and mixins

muhammad-ammar commented on cms/static/js/views/course_video_settings.js:
```diff
@@ -0,0 +1,545 @@
+            return this;
+        },
+
+        setFixedCourseVideoSettingsPane: function() {
```
Instead of setting css, we should try to add/remove classes only. This way rtl/ltr things van be taken care of in css.

mushtaqak replied:
I agree, you are right, I tried that way but couldn't. I am afraid the requirements are such that we have to mix these things here, otherwise it could have been simpler but keeping requirements in mind, I think this the only way we can do.

mushtaqak replied:
And that's why I have added comments explaining too

muhammad-ammar commented on cms/static/js/views/course_video_settings.js:
```diff
@@ -0,0 +1,545 @@
+            this.populateLanguages();
+        },
+
+        addLanguageMenu: function() {
```
Better if we have dropdown in template and then values should be populated on view re-render.

mushtaqak replied:
I have changed this to have a dropdown in template and options updated when needed.

muhammad-ammar commented on cms/static/js/views/course_video_settings.js:
```diff
@@ -0,0 +1,545 @@
+
+            if (self.selectedProvider &&
+                    ((isTurnaroundSelected > 0 && self.selectedProvider === '3PlayMedia') ||
+                    (isTurnaroundSelected  > 0 && isFidelitySelected > 0))) {
```
Can we simplify it?

mushtaqak commented:
@srpearce Addressed your feedback except the re-ordering part because it comes from backend and would take too much time to re-order in front-end.

mushtaqak commented:
@roderickmorales Addressed your feedback except 2 things.

> 1. When the user closes the "settings" div, it should slide out of the view

This was taking a lot time, and probably this pane component would change in near future. That's why I didn't go for immediate change @muzaffaryousaf Is this something we must do ???

> 2. Background color, top and bottom border for languages odd rows need to be updated to reflect the specs (my bad, I missed those specs in the early specs document)

I will probably, add this after the Eid Holidays (I will be hopefully back on Tuesday)

roderickmorales commented:
@mushtaqak @srpearce we need to keep the "course video settings" language because this will eventually have more settings, not just transcripts

roderickmorales commented:
@mushtaqak we need to remove the overflow: scroll from the languages div. The overflow: scroll needs to be on the actual settings div so that I can get to the "save" button if my screen is small.

mushtaqak commented on cms/djangoapps/contentstore/views/videos.py:
```diff
@@ -340,6 +340,9 @@ def transcript_preferences_handler(request, course_key_string):
     provider = data.get('provider', '')
 
+    # TODO: if provider == '': delete course preferences
+    # i.e call delete api end point like delete_transcript_preferences(course_key_string)
```
Yes, we need to implement this endpoint as well. Setting N/A provider would delete the course entry for transcript settings

muzaffaryousaf replied:
@mushtaqak Since you're done with changes, can you start work on this TODO in new branch ?

mushtaqak commented on cms/static/js/views/course_video_settings.js:
```diff
@@ -0,0 +1,546 @@
+/**
+ * CourseVideoSettingsView shows a sidebar containing course wide video settings.
```
Previously it was like that, but after discussing with @muzaffaryousaf names were changed. Primary reason was that these settings can expand other than just transcripts.

muzaffaryousaf replied:
@mushtaqak we decided it to rename it to transcript but let's defer to Sylvia and Roderick to finalize the text. can you tag them both as a reminder, to finalize it.

mushtaqak commented on cms/static/js/views/course_video_settings.js:
```diff
@@ -0,0 +1,546 @@
+
+        initialize: function(options) {
+            var videoTranscriptSettings = options.videoTranscriptSettings;
+            this.activeTranscriptionPlan = options.activeTranscriptPreferences;
```
`activeTranscriptPreferences` are not the part of `videoTranscriptSettings, they are being sent as a different arguments.

mushtaqak commented on cms/static/js/views/course_video_settings.js:
```diff
@@ -0,0 +1,546 @@
+            var self = this;
+
+            // Preventing any parent handlers from being notified of the event. This is to stop from firing the document
+            // level click handler to execute on course video settings pane click.
```
If `course video settings` label is changed, I will change the rest of the occurrences as well. Right now @roderickmorales and @srpearce have to decide this :)

srpearce replied:
@mushtaqak @roderickmorales let me know that this area will eventually have other settings as well, so we can keep this as "course video settings".

roderickmorales replied:
@srpearce yes, this area will eventually grow to have more video related settings

mushtaqak commented on cms/static/js/views/course_video_settings.js:
```diff
@@ -0,0 +1,545 @@
+                            }
+                        )
+                    )
+                );
```
Yes, in `populateLanguages` it being appended to the main container as some languages previously being set, so we need to append them to container

mushtaqak replied:
I have changed this. Now it's not being duplicated.

mushtaqak commented on cms/static/js/views/course_video_settings.js:
```diff
@@ -0,0 +1,545 @@
+            this.selectedLanguages.pop(selectedLanguage);
+        },
+
+        populateProvider: function() {
```
I did this to keep things dynamic if in future we get a new provider, it can easily fit in. @muzaffaryousaf Do you think, we need to make this static?

mushtaqak commented:
> The overflow: scroll needs to be on the actual settings div so that I can get to the "save" button if my screen is small.

This is probably right. I will change right away.
> @mushtaqak we need to remove the overflow: scroll from the languages div.

We need scroll overflow here as well, because many languages can be selected. If I remove overflow:scroll on languages container and only add to the main container.  I get the following screen. See snapshot :
<img width="799" alt="screen shot 2017-09-05 at 5 29 43 pm" src="https://user-images.githubusercontent.com/6991154/30061533-29cc1efa-9261-11e7-8b18-fb541274fb33.png">

mushtaqak commented:
@muhammad-ammar This is expected behaviour:

> unable to see last saved settings after navigating among providers, Steps to reproduce.
select cielo24
add a language
save the settings
select the 3play
select ceilo24 and you will not see settings saved in step 2

Right @muzaffaryousaf ? As this was discussed that changing on provider, all previous user selection would not be considered.

mushtaqak commented:
@muzaffaryousaf This is ready for your review again ?

muzaffaryousaf commented on cms/static/js/spec/views/course_video_settings_spec.js:
```diff
@@ -0,0 +1,312 @@
+                            vi: 'Vietnamese',
+                            'es-419': 'Spanish (Latin America)',
+                            'zh-cmn-Hant': 'Chinese (Traditional)',
+                            pl: 'Polish'
```
It should't be a whole list in tests, can we not use 3,4 languages ?

muzaffaryousaf commented on cms/static/js/spec/views/course_video_settings_spec.js:
```diff
@@ -0,0 +1,312 @@
+                                    hi: 'Hindi',
+                                    ja: 'Japanese',
+                                    es: 'Spanish',
+                                    he: 'Hebrew'
```
Again this should't be a complete list.

muzaffaryousaf commented on cms/static/js/spec/views/course_video_settings_spec.js:
```diff
@@ -0,0 +1,312 @@
+                expect(courseVideoSettingsView.selectedLanguages, activeTranscriptPreferences.preferred_languages);
+            });
+
+            it('saves transcript settings on update settings buttton click if preferances are selected', function() {
```
buttton --> button

muzaffaryousaf commented on cms/static/js/spec/views/course_video_settings_spec.js:
```diff
@@ -0,0 +1,312 @@
+                    error: 'Error message'
+                });
+
+                // Verify that success message is shown.
```
success --> error

muzaffaryousaf commented on cms/static/js/spec/views/course_video_settings_spec.js:
```diff
@@ -0,0 +1,312 @@
+            });
+        });
+    }
+);
```
I'm not able to see TODOs with remaining tests.

muzaffaryousaf commented on cms/static/js/views/course_video_settings.js:
```diff
@@ -0,0 +1,534 @@
+            // Make sticky when scroll reaches top.
+            $(window).scroll(function() {
+                // Remove transition when we start scrolling.
+                // Why we do this? The settings pane does some back and forth movement when it is switched between
```
come back and forth

muzaffaryousaf commented on cms/static/js/views/course_video_settings.js:
```diff
@@ -0,0 +1,534 @@
+
+            // set windows total height
+            $courseVideoSettingsContainer.css('height', windowHeight);
+            $courseVideoSettingsContainer.css('right', 20);
```
This is horrible.

muzaffaryousaf commented on cms/static/js/views/course_video_settings.js:
```diff
@@ -0,0 +1,534 @@
+                    $courseVideoSettingsContainer.css('right', 20);
+                }
+            });
+        },
```
can't we do this by add/remove css classes ?

mushtaqak commented:
@srpearce @roderickmorales can you give final look?

roderickmorales commented:
The overflow in the language list still an issue for me. After six languages the user can't see that a seventh language was added. 
I don't feel like this div should have a scroll.

srpearce commented:
The UI text looks good. Thanks!

mushtaqak commented on cms/static/sass/views/_video-upload.scss:
```diff
@@ -13,6 +13,173 @@
+      .action-close-course-video-settings {
+        width: 100%;
+        padding: ($baseline/2) ($baseline*0.8);
+        text-align: left;
```
I am not sure, should we ?

mushtaqak commented:
@roderickmorales Removed the overflow scroll from languages div. Updated some other css colors too.

@muzaffaryousaf Addressed your feedback. And added transcript metadata to s3 keys and fixed it's test as well. 

@srpearce @roderickmorales @muzaffaryousaf We still need to decide the ultimate text **Course Video Settings** _or_ **Transcript Settings**

srpearce commented:
Course Video Settings is fine.

muzaffaryousaf commented on cms/static/js/views/course_video_settings.js:
```diff
@@ -0,0 +1,537 @@
+/**
+ * CourseVideoSettingsView shows a sidebar containing course wide video settings.
```
video settings which we show on Video Uploads page.

muzaffaryousaf commented on cms/static/js/views/course_video_settings.js:
```diff
@@ -0,0 +1,537 @@
+
+            this.selectedTurnaroundPlan = event.target.value;
+            // Remove any error if present already.
+            this.clearPreferanceErrorState($turnaroundContainer);
```
Typo: `Preferance` --> `Preference`. You're using wrong spelling at several places.

muzaffaryousaf commented on cms/static/js/views/course_video_settings.js:
```diff
@@ -0,0 +1,537 @@
+            this.populateLanguageMenu();
+        },
+
+        populateProvider: function() {
```
populateProvider  --> `renderProviders`. Looks like the purpose of this method is to render the radio buttons with values not populating the values in some existing elements.

muzaffaryousaf commented on cms/static/js/views/course_video_settings.js:
```diff
@@ -0,0 +1,537 @@
+            }
+        },
+
+        populatePreferences: function() {
```
Looks like this is `renderPreferences` since it's not populating the values.

muzaffaryousaf commented on cms/static/js/views/course_video_settings.js:
```diff
@@ -0,0 +1,537 @@
+            this.populateProvider();
+            this.populateTurnaround();
+            this.populateFidelity();
+            this.populateLanguages();
```
Similarly, all of the methods are actually rendering the content rather that just populating the values. can we change them.

mushtaqak commented:
@roderickmorales Need your thumbs up too

mushtaqak commented on cms/static/sass/views/_video-upload.scss:
```diff
@@ -1,3 +1,6 @@
+$gray-background: #ECF0F1;
+$dark-black-blue: #2C3E50;
```
I am not sure if we should add these there as these may not be used somewhere else. Please let me know if you still think I should change.

muzaffaryousaf replied:
They're not being used now but  someone could need them in future just like we used them.

mushtaqak commented:
jenkins run bokchoy

mushtaqak commented:
jenkins run a11y

mushtaqak commented:
jenkins run lettuce
